### ISSUE TITLE
1.27.1-dev125.0 update fails to load extensions in non shell projects

### ISSUE THREAD

**Issue by timdenouden** [duckdb/duckdb-wasm#1425]
### What happens?

In the hosted shell.duckdb.org 0.9.0 shell (wasm v1.27.1-dev125.0) I am able to run `install json` as a no-op and `load json` with success. 

On my local machine I am unable to run `install json` or `load json`, recieving the error: `Error: IO Error: Can't find the home directory at '/home/web_user'
Specify a home directory using the SET home_directory='/path/to/dir' option.`

If it helps, via the ConsoleLogger I get a message: `checkDirectory: /home/web_user` before the error.

### To Reproduce

Test SQL (Works in shell.duckdb.org but fails in local)
```sql
install json;
load json;
SELECT {duck: 42}::JSON;
```

Local setup and run example (bundled with vite)
```typescript
import {
    AsyncDuckDB,
    ConsoleLogger,
    DuckDBBundles,
    selectBundle,
} from "@duckdb/duckdb-wasm";
import duckdb_wasm from "@duckdb/duckdb-wasm/dist/duckdb-mvp.wasm?url";
import mvp_worker from "@duckdb/duckdb-wasm/dist/duckdb-browser-mvp.worker.js?url";
import duckdb_wasm_eh from "@duckdb/duckdb-wasm/dist/duckdb-eh.wasm?url";
import eh_worker from "@duckdb/duckdb-wasm/dist/duckdb-browser-eh.worker.js?url";

const MANUAL_BUNDLES: DuckDBBundles = {
    mvp: {
        mainModule: duckdb_wasm,
        mainWorker: mvp_worker,
    },
    eh: {
        mainModule: duckdb_wasm_eh,
        mainWorker: eh_worker,
    },
};

const bundle = await selectBundle(MANUAL_BUNDLES);
const worker = new Worker(bundle.mainWorker!);
const logger = new ConsoleLogger();
const db = new AsyncDuckDB(logger, worker);
await db.instantiate(bundle.mainModule, bundle.pthreadWorker);

const c = await db.connect();
const t = await c.query("install json; load json; SELECT {duck: 42}::JSON;");
console.log(t.toArray());
```

### Browser/Environment:

Chrome Version 117.0.5938.132 (Official Build) (arm64)

### Device:

Macbook Pro (M1 Pro) macOS 14.0 (Sonoma) 

### DuckDB-Wasm Version:

1.27.1-dev125.0

### DuckDB-Wasm Deployment:

shell.duckdb.org, localhost, JsDelivr CDN

### Full Name:

Timothy Den Ouden

### Affiliation:

Pondabble & Avanade Inc

**Comment by karthik-hr** [comment]
@timdenouden After successfully wasting complete day i realised author haven't published loadable version to npm or jsDelivr. I have downloaded artifacts from here and used js files directly

https://github.com/duckdb/duckdb-wasm/actions/runs/6313814794

**Comment by karthik-hr** [comment]
@timdenouden i think author haven't published loadable version to npm or jsDelivr. I have downloaded artifacts from following link and used **js** and **wasm** files directly and it worked

[duckdb-wasm-packages-loadable]
https://github.com/duckdb/duckdb-wasm/actions/runs/6313814794

**Comment by Youssef-Harby** [comment]
@karthik-hr, Could you provide more details or a step-by-step breakdown of your solution? I'm having trouble implementing it, thanks

**Comment by karthik-hr** [comment]
1. Download required wasm and worker files from latest **main** branch artifacts.(https://github.com/duckdb/duckdb-wasm/suites/16730455912/artifacts/956004323)
2. Extract the file (**packages** will be folder name once you extract artifact)
3. Copy required wasm and worker files to your project
4. import wasm and worker files from your project instead of node_module or jsDelivr

Path for files
Workers: ~/Downloads/**packages**/duckdb-wasm-app/build/release/static/assets
Wasm: ~/Downloads/**packages**/duckdb-wasm-app/build/release/static/wasm

```
import { AsyncDuckDB, ConsoleLogger, DuckDBBundles, selectBundle } from '@duckdb/duckdb-wasm';
import duckdb_wasm from './duckdb-mvp.55a4d22d452f84b16619.wasm';
import duckdb_wasm_eh from './duckdb-eh.fc27ac0d06e691435a6f.wasm';

const MANUAL_BUNDLES: DuckDBBundles = {
	mvp: {
		mainModule: duckdb_wasm,
		mainWorker: new URL('./duckdb-browser-mvp.worker.77ab6c32718967e8b723.js', import.meta.url).toString(),
	},
	eh: {
		mainModule: duckdb_wasm_eh,
		mainWorker: new URL('./duckdb-browser-eh.worker.b5df016aa25ab29299f3.js', import.meta.url).toString(),
	},
};


const bundle = await selectBundle(MANUAL_BUNDLES);
const worker = new Worker(bundle.mainWorker!);
const logger = new ConsoleLogger();
const db = new AsyncDuckDB(logger, worker);
await db.instantiate(bundle.mainModule);

const c = await db.connect();
const t = await c.query('install json; load json;; SELECT {duck: 42}::JSON;");
console.log(t.toArray());

```

**Comment by Youssef-Harby** [comment]
Thank you so much!

**Comment by markhalonen** [comment]
hitting the same issue, would be super wonderful to do an npm deploy if that's all it takes! 🙏

**Comment by tobilg** [comment]
@carlopi will the loadable extensions land in the main releases as well?

**Comment by carlopi** [comment]
Can you give this a try with the latest released package `1.28.1-dev65.0` on [npm](https://www.npmjs.com/package/@duckdb/duckdb-wasm/v/1.28.1-dev65.0) ?

Thanks a lot

**Comment by carlopi** [comment]
I would assume, given https://github.com/duckdb/duckdb-wasm/issues/1542 and https://github.com/duckdb/duckdb-wasm/issues/1561 are solved also this is likely solved by using the latest version: https://www.npmjs.com/package/@duckdb/duckdb-wasm/v/1.28.1-dev91.0.

If you can give this a try again and confirm, it would be amazing.
Thanks a lot for raising this in the first place.
